warehouse/manage/forms.py

```diff
@@ -713,6 +716,23 @@ class CreateTeamForm(SaveTeamForm):
     __params__ = SaveTeamForm.__params__
 
 
+def validate_is_simple(form, field):
+    try:
+        timeout = Timeout(connect=1.0, read=1.0)
+        http = PoolManager(timeout=timeout)
+        response = http.request(
+            "HEAD", field.data, headers={"Accept": ", ".join(MIME_PYPI_SIMPLE_V1_ALL)}
+        )
+        if response.headers.get("Content-Type") not in MIME_PYPI_SIMPLE_V1_ALL:
+            raise wtforms.validators.ValidationError(
+                _("Unable to parse simple index at given url")
+            )
+    except Exception as exc:
+        raise wtforms.validators.ValidationError(
+            _(f"Unable to parse simple index at given url: {exc}")
```

Comment:
```suggestion
            _(f"The URL provided does not respond as a Simple repository: {exc}")
```

And maybe even tack on a URL? https://packaging.python.org/specifications/simple-repository-api/